### PR TITLE
fix(curriculum): prevent hardcoding pokemon

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Ref #54063

Two tests were added. One randomly generates an invalid Pokemon name and tests for a Pokemon not found response. The other test randomly generates a valid Pokemon id and compares it to the types received from our Pokemon API and sees if they match. If I had just used a few regular names, we would have to continually add new options to our generic tests. It also makes test writing a lot smoother.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-pokemon-search-app-project/build-a-pokemon-search-app.md [hunk 1]
```diff
@@ -345,6 +345,83 @@ async () => {
...
+      assert.include(['pokémon not found', 'pokemon not found'], alertMessage.trim().replace(/[.,?!]+$/g, '').toLowerCase());
+    }
+  } catch (err) {
+    throw new Error(err);
+  }
+};
+```
+
+
+When the `#search-input` element contains an valid Pokemon id and the `#search-button` element is clicked, an alert should appear with the correct type.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-pokemon-search-app-project/build-a-pokemon-search-app.md:390]:
```suggestion
When the `#search-input` element contains a valid Pokemon id and the `#search-button` element is clicked, the UI should be filled with the correct data.
```

**Review by majestic-owl448 — COMMENTED** [review]
I am living a couple of wording suggestions, I never completed this project so I don't have a project to test these new tests with.

**Comment by a2937** [comment]
> I am living a couple of wording suggestions, I never completed this project so I don't have a project to test these new tests with.

Except there is a solution attached at the bottom of the file you can test against.

**Review by Ksound22 — APPROVED** [review]
LGTM 🚀

**Review by JoyShaheb — APPROVED** [review]

**Pull request merged**